Assets/DragDrop.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public enum DropType
+{ 
+    None,
+    Head,
+    Hand,
+    Foot,
+    Heart
+}
+
+
+public class DragDrop : MonoBehaviour, IPointerDownHandler, IBeginDragHandler, IEndDragHandler, IDragHandler
+{
+    public RectTransform rectTransform;
+    public DropType dropType = DropType.None;
+
+    private Transform lastPosition;
+    private Canvas canvas;
+    private CanvasGroup canvasGroup;
+
+    void Start()
+    {
+        canvas = PlanningSceneManager.Instance.gameCanvas;
+        rectTransform = GetComponent<RectTransform>();
+        canvasGroup = GetComponent<CanvasGroup>();
+    }
+
+    public void OnPointerDown(PointerEventData eventData)
+    {
+        //Not Implemented
+    }
+
+    public void OnBeginDrag(PointerEventData eventData)
+    {
+        transform.parent.GetComponent<DropSlot>().ClearSlot();
+        lastPosition = transform.parent;
+        transform.SetParent(canvas.transform, true);
+        transform.SetAsLastSibling();
+        canvasGroup.alpha = 0.6f;
```

Comment:
Are these things that might be useful to have available in the editor to tweak?

Reply:
these ones in particular I don't wanna expose to the editor!